gateway/utils/syncman/syncman_crud.go

```diff
@@ -87,6 +87,109 @@ func (s *Manager) RemoveDatabaseConfig(ctx context.Context, project, dbAlias str
 	return s.setProject(ctx, projectConfig)
 }
 
+// GetPreparedQuery gets preparedQuery from config
+func (s *Manager) GetPreparedQuery(ctx context.Context, project, dbAlias, id string) ([]interface{}, error) {
+	// Acquire a lock
+	type response struct {
+		ID        string   `json:"id"`
+		SQL       string   `json:"sql"`
+		Arguments []string `json:"arguments" yaml:"arguments"`
+	}
+	s.lock.Lock()
+	defer s.lock.Unlock()
+
+	projectConfig, err := s.getConfigWithoutLock(project)
+	if err != nil {
+		return nil, err
+	}
+
+	if dbAlias != "" {
+		databaseConfig, ok := projectConfig.Modules.Crud[dbAlias]
+		if !ok {
+			return nil, fmt.Errorf("specified database:{%s} not present in config", dbAlias)
+		}
+
+		if id != "" {
+			preparedQuery, ok := databaseConfig.PreparedQueries[id]
+			if !ok {
+				return nil, fmt.Errorf("Prepared Queries for id (%s) not present in config for dbAlias (%s) )", id, dbAlias)
+			}
+			return []interface{}{&response{ID: preparedQuery.ID, SQL: preparedQuery.SQL, Arguments: preparedQuery.Arguments}}, nil
+		}
+		preparedQuery := databaseConfig.PreparedQueries
+		var coll []interface{} = make([]interface{}, 0)
+		for key, value := range preparedQuery {
+			coll = append(coll, &response{ID: key, SQL: value.SQL, Arguments: value.Arguments})
+		}
+		return []interface{}{coll}, nil
```

Comment:
Did you even once try consdering why we always return an array with a single element?
```suggestion
		return coll, nil
```